PocketKit/Sources/Analytics/Events/System.swift

```diff
@@ -25,12 +25,23 @@ public struct System: Event, CustomStringConvertible {
         }
     }
 
+    public var value: String? {
+        switch type {
+        case .userMigration(let userMigrationState):
+            return userMigrationState.id(source)
+        }
+    }
+
     public func toSelfDescribing() -> SelfDescribing {
-        let base = SelfDescribing(schema: System.schema, payload: [
+        var payload = [
             "identifier": NSString(string: self.description),
-        ])
+        ]
 
-        return base
+        if let value = value {
+            payload["value"] = NSString(string: value)
```

Comment:
@CMasterson I added this, given we have this field in snowplow we should use it to capture error descriptions if they exist to help inform how we proceed.

Reply:
Totes, seems like an oversight on my part.